### PR TITLE
Add support for MobileNet v2 in Ethos-U55 and arm baremetal runner

### PR DESCRIPTION
**Pull request by robell**
- Builds on TOSA support for mv2
- Currently has accuracy issues for on-target execution.
- Adding support mv2 from examples/models (torchvision mv2)
- Adds ability to use portable Q/DQ ops to enable float signature quantized models (to be improved with int signature when possible to enable integer images rather than int -> float -> quant flow.
- Adds permute for ethos-u backend to enable NCHW to NHWC (which we'll aim to replace with a compiler pass on entry/exit of the subgraph, or the NHWC support when available.
- small fix to enable qantized ops in runtime for the IOQDQ pass, and build/register those kernels/ops in bare metal build

### CODE REVIEW

File: backends/arm/runtime/ArmBackendEthosU.cpp [hunk 1]
```diff
@@ -108,18 +103,93 @@ class ArmBackend final : public PyTorchBackendInterface {
...
+              handles.inputs->io[i].elem_size == 4 or
+          tensor_in.scalar_type() == ScalarType::Char and
+              handles.inputs->io[i].elem_size == 1) {
+        if (permuted_input_shape && tensor_in.dim() == 4) {
+          // permuted copy CHW to HWC
+          char* input = scratch_addr;
+          char* output = tensor_in.mutable_data_ptr<char>();
+          int H = tensor_in.size(1);
+          int W = tensor_in.size(1);
+          for (int i = 0; i != H * W; ++i) {
```
Comment (digantdesai) [inline backends/arm/runtime/ArmBackendEthosU.cpp:178]: oh man, this can be slow, no?

File: backends/arm/runtime/ArmBackendEthosU.cpp [hunk 2]
```diff
@@ -108,18 +103,93 @@ class ArmBackend final : public PyTorchBackendInterface {
...
+      if (!supported) {
+        ET_LOG(
+            Error,
+            "Input %d expected Integer (4 byte) or Char (1 byte) integer inputs",
+            i);
+        return Error::InvalidProgram;
+      }
+
+      // Special case implicitly permuted rank4 tensors
+      int permuted_input_shape = false;
```
Comment (digantdesai) [inline backends/arm/runtime/ArmBackendEthosU.cpp:133]: nit: can we refactor the permute stuff out as a function?

File: examples/arm/run.sh [hunk 3]
```diff
@@ -131,8 +139,9 @@ function run_fvp() {
         -C mps3_board.visualisation.disable-visualisation=1 \
         -C mps3_board.telnetterminal0.start_telnet=0        \
         -C mps3_board.uart0.out_file='-'                    \
+        -C mps3_board.uart0.shutdown_on_eot=1               \
         -a "${elf}"                                         \
-        --timelimit 5 || true # seconds
+        --timelimit 120 || true # seconds
```
Comment (digantdesai) [inline examples/arm/run.sh:144]: woah, portable q/dq are slow I guess on FVP :p

**Review by digantdesai — APPROVED** [review]
This is great. Thanks @robell.

File: backends/arm/runtime/ArmBackendEthosU.cpp (hunk 2 above)
Comment (robell) [reply backends/arm/runtime/ArmBackendEthosU.cpp]: Sure can! ;)

File: examples/arm/run.sh (hunk 3 above)
Comment (robell) [reply examples/arm/run.sh:144]: not that slow, this is just a backstop time as the program now exits much sooner with ascii 0x4 (EOT) and the shutdown_on_eot flag above, so the timelimit is just a large one for unclean exit.

**Comment by digantdesai** [comment]
Sigh.. can you rebase please?

**Comment by robell** [comment]
@digantdesai or @mergennachin could we import this now, thanks to the fix in #3449 this now passes when rebased.

File: backends/arm/runtime/ArmBackendEthosU.cpp (hunk 2 above)
Comment (robell) [reply backends/arm/runtime/ArmBackendEthosU.cpp]: this was refactored out.

File: examples/arm/run.sh (hunk 3 above)
Comment (robell) [reply examples/arm/run.sh:169]: fwiw, the mv2 emulation run is about 10 seconds on a single modern core, and exits at that point with the EOT

File: backends/arm/arm_partitioner.py [hunk 4]
```diff
@@ -64,6 +68,54 @@ def is_node_supported_custom(self, node: torch.fx.Node) -> bool:
         return True
 
 
+from executorch.exir.pass_base import ExportPass, PassResult
+from executorch.exir.passes import PassManager
+
+
+class TagIOQuant(ExportPass):
+    """
+    Pass run before partitioning to tag Q/DQ on any placeholder and output
```
[the remark this answers is not in the document]
Comment (robell) [reply backends/arm/arm_partitioner.py:78]: resolved this and removed the tag.

File: backends/arm/arm_backend.py [hunk 5]
```diff
@@ -104,6 +105,10 @@ def set_permute_memory_format(self, set_nhwc_permutation: bool = True):
         self.permute_nhwc = set_nhwc_permutation
         return self
 
+    def set_quantize_io(self, quantize_io: bool = False):
```
[the remark this answers is not in the document]
Comment (robell) [reply backends/arm/arm_backend.py:112]: we've a few other cases like this, i've asked the original author to change.

**Pull request closed without merging**